### PR TITLE
getMaxTradeSize(), getLiquidationPrice() and uPnL overhaul

### PR DESCRIPTION
**Pull request by filipzeta**
- getMaxTradeSize() - figures out the maximum size you can trade based on if you're closing or opening a position.
- getLiquidationPrice() - figures out at what price your position will get liquidated.
- Split uPnL calcs into estimateRealizedPnl() and calculateUnrealizedPnl()
- checkLiquidity() - finds the best price to execute an order at

### CODE REVIEW

**Review by MeanMangosteen — APPROVED** [review]

**Comment by filipzeta** [comment]
- [ ] Split uPnL calc with executionPrices into a separate function for estimated realized PnL

File: src/utils.ts [hunk 1]
```diff
@@ -1710,3 +1710,81 @@ export function median(arr: number[]): number | undefined {
...
+  // Invalid order
+  if (!orderbookSide || !orderbookSide.length) {
+    return { validLiquidity: false, avgPrice: 0, worstPrice: 0 };
+  }
+
+  let validLiquidity = false;
+  // Size seen on the orderbook that satisfies the specified price
+  let seenSize = 0;
+  // The cumulative trade value for the seen size
+  let cumAmount = 0;
```
Comment (MeanMangosteen) [inline src/utils.ts:1734]: I find this highly inappropriate.

File: src/utils.ts [hunk 2]
```diff
@@ -1710,3 +1710,81 @@ export function median(arr: number[]): number | undefined {
   const mid = Math.floor(s.length / 2);
   return s.length % 2 === 0 ? (s[mid - 1] + s[mid]) / 2 : s[mid];
 }
+
+export const checkLiquidity = (
+  size: number,
+  asset: Asset,
+  orderbook?: types.DepthOrderbook
```
Comment (MeanMangosteen) [inline src/utils.ts:1717]: Given the logic, it makes more sense to make this required now. It's needed to get the liquidity whereas in the FE there was a fallback value for orderbook if it wasn't provided.

File: src/utils.ts (hunk 1 above)
Comment (filipzeta) [reply src/utils.ts:1734]: 🤓

File: src/utils.ts (hunk 2 above)
Comment (filipzeta) [reply src/utils.ts]: yep good point, ive set it to just be `orderbook:` now

**Review by MeanMangosteen — APPROVED** [review]

**Pull request merged**